pipeline-modules/deployment-service/gcp/istio-gke-regional.yaml

```diff
@@ -479,17 +483,17 @@ template: |
     - '-c'
     - |
       cd {{helm_chart_path}} ;
-      helm upgrade --install -f values.yaml {{applicationName}}-{{environment}} . --atomic --timeout {{deployment_timeout}}
-      kubectl rollout restart deployment {{applicationName}}-deployment -n {{applicationName}}
-      kubectl rollout status deployment {{applicationName}}-deployment   -n {{applicationName}}
+      helm upgrade --install -f values.yaml {{service_name}} . --atomic --timeout {{deployment_timeout}}
+      kubectl rollout restart deployment {{fmt_app_name}}-deployment -n {{fmt_app_name}}
+      kubectl rollout status deployment {{fmt_app_name}}-deployment   -n {{fmt_app_name}}
```

Comment:
I hope it doesn't affect our existing deployments on staging, prod and RSAF Prod. 

The trimming of names might cause an issue (or is it the helm/kubectl limit?

Reply:
It will not affect the existing deployments as the existing names (app, env etc) are already compliant with the kind of formatting we are doing.

As part of VAN-3766 we have another PR in vanguard-api which removes the existing validations on names. https://github.com/cldcvr/vanguard-api/pull/1342

Till now the names given by users were passing the validations, once we remove these validations, we may observe some names which would require formatting. So the filter here takes care of that.

Reply:
@cc-shekher  if our ApplicationName is **external-events-service** , than what will be the service_name values

Reply:
I believe you refer to the applicationName. As service_name gets derived from the combination of applicationName and environment name.

if application name is `external-events-service` then the derived name which is fmt_app_name will be same. As it doesn't contain any special char.